### PR TITLE
[#899] Docs: fix broken link and update outdated info

### PR DESCRIPTION
**Pull request by Tejas2805**
* [x] Fix broken link to group-config.csv in User Guide #899 
* [x] Update outdated info in User Guide #891 
* [x] Update outdated info in Developer Guide 

```
The link to an example group-config.csv in the user guide 
is broken. There is outdated info in the User Guide and the 
Developer Guide. 

The link is broken because of the file, group-config.csv,
doesn't exist where the link states it to be.

Let's fix this by adding the file, group-config.csv, in the 
correct path. 
Let's update the User Guide and Developer Guide to represent
the current stage. 
```

### CODE REVIEW

**Comment by Tejas2805** [comment]
@yong24s Issue #891 also relates to User Guide? If you think that issue needs to be worked on too, I will make the change in this PR itself. Since this also relates to UG. Let me know your thoughts on it.

**Comment by yong24s** [comment]
@Tejas2805 

Since the fix is trivial, lets do it in this PR.

**Comment by Tejas2805** [comment]
@eugenepeh @fzdy1914 Ready for review.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ gradlew run -Dargs="-c ./configs/ -o output_path/ -s 21/10/2017 -u 21/11/2017 -f
 ### Parser(ConfigParser)
 [`Parser`](/src/main/java/reposense/parser) contains three components:
  * [`ArgsParser`](/src/main/java/reposense/parser/ArgsParser.java): Parses the user-supplied command line arguments into a `CliArguments` object.
- * [`CsvParser`](/src/main/java/reposense/parser/CsvParser.java): Abstract generic class for CSV parsing functionality. The following two classes extends `CsvParser`.
+ * [`CsvParser`](/src/main/java/reposense/parser/CsvParser.java): Abstract generic class for CSV parsing functionality. The following three classes extend `CsvParser`.
```
Comment (eugenepeh) [inline docs/DeveloperGuide.md]: this is a dev guide change; nothing to do with user guide

File: docs/UserGuide.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Here is an example of how the report looks like:
 
 ![report](images/report-features.png)
 
-It consists of three main parts: the [_Chart Panel_](#chart-panel), the [_Code Panel_](#code-panel), and the [_Tool Bar_](#tool-bar), each of which is explained in the sections below.
+It consists of four main parts: the [_Chart Panel_](#chart-panel), the [_Code Panel_](#code-panel), the [_Commits Panel_](#commits-panel), and the [_Tool Bar_](#tool-bar), each of which is explained in the sections below.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: this doesnt seems to be related to the PR title

**Comment by jamessspanggg** [comment]
@Tejas2805 please ensure that your commit message body is within 72 characters

**Comment by eugenepeh** [comment]
@reposense/devs do help to take a look if the docs have any relevant components that need to be updated.

**Pull request merged**